pwndbg/next.py

```diff
@@ -103,6 +108,18 @@ def break_next_call(symbol_regex=None):
         if ins.symbol and re.match('%s$' % symbol_regex, ins.symbol):
             return ins
 
+
+def break_next_ret(address=None):
+    while pwndbg.proc.alive:
+        ins = break_next_branch(address)
+
+        if not ins:
+            break
+
+        if capstone.CS_GRP_RET in ins.groups:
```

Comment:
Yeah, `nextsc` and friends all work fine for me.

```
$ git log --oneline -1
0bd903e Workaround for gdb remote target search bug described in #321 (#322)
$ pwn shellcraft arm.linux.sh -felf > shellcode
$ pwn debug --exe shellcode
```

```
...
pwndbg> nextsc
...
LEGEND: STACK | HEAP | CODE | DATA | RWX | RODATA
[──────────────────────────────────────────────REGISTERS──────────────────────────────────────────────]
*R0   0xf6ffed94 ◂— '/bin///sh'
*R1   0xf6ffed88 —▸ 0xf6ffed90 ◂— 0x6873 /* 'sh' */
 R2   0x0
 R3   0x0
 R4   0x0
 R5   0x0
 R6   0x0
*R7   0xb
 R8   0x0
 R9   0x0
 R10  0x10000000 ◂— movw   r7, #0x68 /* 0xe3007068; 'hp' */
 R11  0x0
*R12  0xf6ffed90 ◂— 0x6873 /* 'sh' */
*SP   0xf6ffed88 —▸ 0xf6ffed90 ◂— 0x6873 /* 'sh' */
*PC   0x10000054 ◂— svc    #0 /* 0xef000000 */
[───────────────────────────────────────────────DISASM────────────────────────────────────────────────]
   0x10000040    mov    ip, r1
   0x10000044    str    ip, [sp, #-4]!
   0x10000048    mov    r1, sp
   0x1000004c    eor    r2, r2, r2
   0x10000050    mov    r7, #0xb
 ► 0x10000054    svc    #0 <SYS_execve>
        path: 0xf6ffed94 ◂— '/bin///sh'
        argv: 0xf6ffed88 —▸ 0xf6ffed90 ◂— 0x6873 /* 'sh' */
        envp: 0x0
   0x10000058    andeq  r1, r0, r1, asr #24

[────────────────────────────────────────────────STACK────────────────────────────────────────────────]
00:0000│ r1 sp  0xf6ffed88 —▸ 0xf6ffed90 ◂— 0x6873 /* 'sh' */
01:0004│        0xf6ffed8c ◂— 0x0
02:0008│ r12    0xf6ffed90 ◂— 0x6873 /* 'sh' */
03:000c│ r0     0xf6ffed94 ◂— '/bin///sh'
04:0010│        0xf6ffed98 ◂— '///sh'
05:0014│        0xf6ffed9c ◂— 0x41410068 /* 'h' */
06:0018│        0xf6ffeda0 ◂— 0x1
07:001c│        0xf6ffeda4 —▸ 0xf6ffef1d ◂— 0x6d6f682f ('/hom')
[──────────────────────────────────────────────BACKTRACE──────────────────────────────────────────────]
 ► f 0 10000054
Breakpoint *0x10000054
pwndbg>
```

A small bug that I do notice is that the `context` is displayed twice.